doc/docs/server-pull.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Pull Endpoint
+slug: /server-pull
+---
+
+The pull endpoint is used when doing a [`pull`](api/classes/replicache#pull).
+
+The purpose of the server pull endpoint is to send the client view to the
+client. The client view is the view of the data that the client should see.
+
+This is an HTTP POST to a URL configured by the
+[`pullURL`](api/interfaces/replicacheoptions#pullurl).
+
+```js
+const rep = new Replicache({
+  pullURL: '/replicache-pull',
+});
+```
+
+Then when the pull happens Replicache will do an HTTP POST request:
+
+```http
+POST /replicache-pull HTTP/2
+```
+
+Replicache also sends the following HTTP request headers:
+
+## HTTP Request Headers
+
+```http
+Content-type: application/json
+Authorization: <auth>
+X-Replicache-RequestID: <request-id>
+```
+
+### `Content-type`
+
+The POST body of the request is encoded as [JSON](https://www.json.org/json-en.html).
+
+### `Authorization`
+
+This is a string that can be used to authorize a user. The auth token can be set
+by defining [`getPullAuth`](api/classes/replicache#getpullauth). This function is
+called when a request returns an HTTP `401 Unauthorized` and can be used to show
+a login screen or request a new auth token as needed.
+
+### `X-Replicache-RequestID`
+
+The request ID is useful for debugging. It is of the form
+`<clientid>-<sessionid>-<request count>`. The request count enables one to find
+the request following or preceeding a given request. The sessionid scopes the
+request count, ensuring the request id is probabilistically unique across
+restarts (which is good enough).
+
+This header is useful when looking at logs to get a sense of how a client got to
+its current state.
+
+## HTTP Request Body
+
+When pulling we `POST` an HTTP request with a JSON encoded body.
+
+```ts
+type PullRequestBody = {
+  clientID: string;
+  cookie: JSONValue;
+  lastMutationID: number;
+  pullVersion: number;
+  schemaVersion: string;
+};
+```
+
+### `clientID`
+
+The [`clientID`](api/classes/replicache#clientid) for this instance of
+Replicache. Each web browser and instance of Replicache gets a unique client ID
+keyed by the [`name`](api/interfaces/replicacheoptions#name).
+
+### `cookie`
+
+The cookie that was received last time a pull was done. `null` if this is the first time we do a pull.
+
+<!-- TODO: Is this null the first time or is the property missing? -->
+
+### `lastMutationID`
+
+When doing a pull this is the ID of th last mutation that was applied to the client.
```

Comment:
@phritz why do we send `lastMutationID` in pull? What are servers expected to do with it?